(#17) Convert issue templates to issue forms

AdmiringWorm opened the pull request:
## Description Of Changes

This pull request updates the existing issue templates that we are using to instead be using the new issue forms instead.
Some structural changes, and added description of earch section has also been added.

## Motivation and Context

There are several times that a user decides to instead of using the templates that we provide, to instead clear out all 
information and going down their own route. While changing this to issue forms do not make such a thing impossible, it do make 
it harder for a user to circumvent filling out the information we need.


## Testing 

Not yet tested, but will be tested on my fork before PR leaves draft state.

## Change Types Made
<!-- Tick the boxes for the type of changes that have been made -->

* [ ] Bug fix (non-breaking change)
* [x] Feature / Enhancement (non-breaking change)
* [ ] Breaking change (fix or feature that could cause existing functionality to change)
* [ ] PowerShell code changes.

## Related Issue
<!-- Make sure you have raised an issue for this pull request before
continuing. -->

Fixes #17

## Change Checklist

* [ ] Requires a change to the documentation
* [x] Documentation has been updated
* [ ] Tests to cover my changes, have been added
* [ ] All new and existing tests passed.
* [ ] PowerShell v2 compatibility checked.

pauby commented on .github/ISSUE_TEMPLATE/zReportIssue.yml:
```diff
@@ -0,0 +1,117 @@
+      value: |
+        Ensure you have read over [Submitting Issues](https://github.com/chocolatey/.github/blob/main/SUBMITTING-ISSUES.md) and you are opening this issue in the correct repository.
+
+        Please check to see if your problem already exists with a quick search of the issues. Start with one relevant term and then add if you get too many results.
```
```suggestion
        Please check to see if your problem already exists with a quick search of the issues. Start with one relevant term and then add others if you get too many results.
```

pauby commented on .github/ISSUE_TEMPLATE/zReportIssue.yml:
```diff
@@ -0,0 +1,117 @@
+        Name your issue appropriately: give it a sentence that reads well enough for anyone seeing this in the release notes to know what it is.
+
+        When writing out the issue details, please ensure you are writing it as if you were explaining it to somebody else, even if you will be working on and resolving the issue yourself.
+        This helps others to understand the reasons for the pull request and for it to be searchable in the future.
```
```suggestion
        This helps others to understand the reasons for the issue and for it to be searchable in the future.
```

pauby commented on .github/ISSUE_TEMPLATE/zReportIssue.yml:
```diff
@@ -0,0 +1,117 @@
+        * Operating System: To get this information run `[System.Environment]::OSVersion.version.tostring()`
+        * Windows PowerShell Version: Run `$PSVersionTable`
+        * Chocolatey CLI Version: Run `choco --version`
+        * Terminal/Emulator: Some knowledge from you are necessary here. CMD, Windows PowerShell, PowerShell Core, Microsoft Terminal, Cmder, and more are possible known terminals or emulators.
```
```suggestion
        * Terminal/Emulator: Some knowledge from you is necessary here. CMD, Windows PowerShell, PowerShell Core, Microsoft Terminal, Cmder, and more are possible known terminals or emulators.
```

pauby commented on .github/ISSUE_TEMPLATE/zReportIssue.yml:
```diff
@@ -0,0 +1,117 @@
+      label: Installed Packages
+      description: |
+        Sometimes one or more packages installed may be the cause of the problem. Please include the information from the command `choco list -lo`. If this is part of the installation of Chocolatey CLI or no packages
+        is installed (not even Chocolatey); please add `N/A` instead.
```
```suggestion
        are installed (not even Chocolatey); please add `N/A` instead.
```

pauby commented on .github/ISSUE_TEMPLATE/zReportIssue.yml:
```diff
@@ -0,0 +1,117 @@
+    attributes:
+      label: Output Log
+      description: |
+        Please include the log given by Chocolatey CLI when running in debugging and verbose mode. You can run Chocolatey CLI in debugging and verbose mode using the flags `--debug --verbose`. Alternatively, you can find the log file in the following default location: `C:\ProgramData
+        \chocolatey\logs\chocolatey.log`.
+        If opening an issue for Chocolatey GUI, please include the information from the Chocolatey GUI log file in: `C:\ProgramData\Chocolatey GUI\Logs`. These files are date based, and you are most likely interested in the latest log file created if the bug happened the last time you launched Chocolatey GUI.
```
```suggestion
        Please include the log given by Chocolatey CLI when running in debugging and verbose mode. You can run Chocolatey CLI in debugging and verbose mode using the flags `--debug --verbose`. Alternatively, you can find the log file in the following default location: `C:\ProgramData\chocolatey\logs\chocolatey.log`.
```

pauby commented on .github/ISSUE_TEMPLATE/zReportIssue.yml:
```diff
@@ -0,0 +1,117 @@
+        If opening an issue for Chocolatey GUI, please include the information from the Chocolatey GUI log file in: `C:\ProgramData\Chocolatey GUI\Logs`. These files are date based, and you are most likely interested in the latest log file created if the bug happened the last time you launched Chocolatey GUI.
+
+        Before including the output log, make sure of the following:
+        - Make sure there is no sensitive data shared.
```
```suggestion
        - There is no sensitive data shared in the log.
```

pauby commented on .github/ISSUE_TEMPLATE/zReportIssue.yml:
```diff
@@ -0,0 +1,117 @@
+        - We need ALL OUTPUT, not just what you may believe is relevant.
+        - We need ALL OUTPUT (including the configuration information). See https://gist.github.com/ferventcoder/b1300b91c167c8ac8205#file-error-txt-L1-L41 for what we need.
+        - If it is hard to reproduce with debug/verbose, the log file already logs with those parameters. Just grab the relevant section from the log file.
+        - Prefer to link to a gist instead of pasting the entire log in the issue. You can create a new gist here: https://gist.github.com/ (see https://docs.github.com/en/github/writing-on-github/editing-and-sharing-content-with-gists/creating-gists).
```
```suggestion
        - Preferably link to a gist instead of pasting the entire log in the issue. You can create a new gist here: https://gist.github.com/ (see https://docs.github.com/en/github/writing-on-github/editing-and-sharing-content-with-gists/creating-gists).
```

pauby commented on .github/ISSUE_TEMPLATE/zReportIssue.yml:
```diff
@@ -0,0 +1,117 @@
+    attributes:
+      label: Additional Context
+      description: |
+        Please include the information in this location if you have other information that may be relevant to the problem you are seeing but is not covered by different sections.
```
```suggestion
        Please include any other information here that you feel may be relevant to the problem you are seeing but is not covered in the previous sections.
```

pauby reviewed (changes requested):
This is awesome @AdmiringWorm. Thanks for doing this. Just a few minor updates.

AdmiringWorm commented:
@pauby I have updated as part of your review, converted the feature/enhancement issue template to form and fixed a typo in the bug issue configuration.

pauby reviewed (approved):
LGTM!

corbob reviewed (approved):
Apparently I've had this tab sitting open for a few weeks. This looks good to me 👍

AdmiringWorm commented:
Moving to draft until it is updated with similar changes made in #22

pauby commented:
I've made those changes. Once merged, #22 can be closed.
